design/database.md

```diff
@@ -0,0 +1,48 @@
+# Database Design
+
+## Table: pipeline\_groups
+| Field name | Data type | Extra
+| --- | --- | ---
+| id | integer | `primary_key`
+| name | string | `NOT NULL`
+| inserted\_at | datetime |
+| updated\_at | datetime |
+
+## Table: pipelines
+| Field name | Data type | Extra
+| --- | --- | ---
+| id | integer | `primary_key`
+| name | string | `NOT NULL`
+| pipeline\_group\_id | belongs\_to | `NOT NULL`
+| inserted\_at | datetime |
+| updated\_at | datetime |
+
+## Table: pipelines\_materials
+| Field name | Data type | Extra
+| --- | --- | ---
+| pipeline\_id | integer | `compound_key`
+| material\_id | integer | `compound_key`
+
+## Table: materials
+| Field name | Data type | Extra
+| --- | --- | ---
+| id | integer | `primary_key`
+| type | string | `enum: {git|file|...}`
+| url | string | `{git://...|file://source_pipeline/release.tar.gz}`
+| inserted\_at | datetime |
+| updated\_at | datetime |
+
+## Table: environment\_variables
+| Field name | Data type | Extra
+| --- | --- | ---
+| id | integer | `primary_key`
+| name | string
+| value | text
+| inserted\_at | datetime |
```

Comment:
Should we add a boolean `secret` field for variables that should be hidden? (Like passwords?)

Reply:
I'll add the secret field, I was hoping to just order alphabetically all the time!

Reply:
I think alphabetically is fine and should work out most of the time if not all!

Worst case scenario we implement visual ordering on a per user bases browser side.